python/graphstorm/model/node_decoder.py

```diff
@@ -65,10 +65,33 @@ def forward(self, inputs):
 
     def predict(self, inputs):
         """ Make prediction on input data.
+
+        Parameters
+        ----------
+        inputs : tensor
+            The input features
+
+        Returns
+        -------
+        Tensor : maximum of the predicted results
+        """
+        logits = th.matmul(inputs, self.decoder)
+        return logits.argmax(dim=1)
+
+    def predict_proba(self, inputs):
+        """ Make prediction on input data.
+
+        Parameters
+        ----------
+        inputs : tensor
+            The input features
+
+        Returns
+        -------
+        Tensor : all the predicted results
         """
         logits = th.matmul(inputs, self.decoder)
```

Comment:
technically, we shouldn't return just the logits here.
we should apply softmax if it's not multilabel and apply sigmoid if it is multilabel

Reply:
the reason why is that logits themselves are unnormalized. customer would expect normalized probability estimates.